astropy/io/ascii/daophot.py

```diff
@@ -1,26 +1,99 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """An extensible ASCII table reader and writer.
-
 daophot.py:
   Classes to read DAOphot table format
-
 :Copyright: Smithsonian Astrophysical Observatory (2011)
 :Author: Tom Aldcroft (aldcroft@head.cfa.harvard.edu)
 """
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import re
 import numpy as np
+import itertools as itt
+from collections import defaultdict
+
 from . import core
 from . import basic
 from . import fixedwidth
 from ...utils import OrderedDict
+from ...extern.six.moves import zip, map
+from .misc import first_true_index, first_false_index, groupmore
 
 
 class DaophotHeader(core.BaseHeader):
-    comment = r'\s*#K'
-    aperture_values = ''
+    """Read the header from a file produced by the IRAF DAOphot routine."""
+    def __init__(self):
+        core.BaseHeader.__init__(self)
+        self.comment = r'\s*#K'
+
+        #self.re_digit_extract = re.compile( '\d+' )
+        self.re_format = re.compile( '%-?(\d+)\.?\d?[sdfg]' )                        #regex for extracting the format strings
+        self.re_header_keyword = re.compile(r'[#]K'
+                                            r'\s+ (?P<name> \w+)'
+                                            r'\s* = (?P<stuff> .+) $',
+                                            re.VERBOSE)
+
+        self.aperture_values = ()
+
+    def __str__():
```

Comment:
I found it useful mostly only when I was debugging the code to check whether the header lines where being read correctly.  The info in the `meta` attribute is hard to read at a glance because of the multiply nested dictionaries.  So this was a kind of intermediary.  I think it may be useful to add this to the BaseHeader - that way one could  more easily check the content of the header.